crypto/src/main/scala/org/bitcoins/crypto/Sign.scala

```diff
@@ -51,6 +51,11 @@ trait Sign {
       entropy: ByteVector): Future[ECDigitalSignature] =
     signWithEntropyFunction(bytes, entropy)
 
+  def schnorrSignFunction: ByteVector => Future[SchnorrDigitalSignature]
```

Comment:
This really feels like it should be some sort of ADT as a subclass of `Sign`. Then we can have a `ECDSASign` and a `SchnorrSign`. Thoughts?

Reply:
NACK from me. There shouldn't be anything that can sign for ecdsa and not schnorr or vice versa.